From adf5f2833517758152cbc9032dd93934a1e16ca1 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Mon, 11 May 2020 11:55:45 +0100
Subject: [PATCH] overlays: tc358743: Use intra-overlay fragments

The tc358743 overlay was written using a workaround to a problem with
fragments that target other fragments, but this had the unfortunate
side-effect of preventing the overlay from being applied at runtime
(the kernel doesn't allow nodes to be overwritten by an overlay, only
properties).

The current firmware and dtoverlay/dtparam utilities include support
for these "intra-overlay" fragments, so remove the workaround and do
it properly.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 .../boot/dts/overlays/tc358743-overlay.dts    | 20 ++++---------------
 1 file changed, 4 insertions(+), 16 deletions(-)

--- a/arch/arm/boot/dts/overlays/tc358743-overlay.dts
+++ b/arch/arm/boot/dts/overlays/tc358743-overlay.dts
@@ -54,28 +54,16 @@
 	};
 
 	fragment@2 {
-		target = <&i2c_csi_dsi>;
+		target = <&tc358743>;
 		__overlay__ {
-			tc358743@0f {
-				port {
-					endpoint {
-						data-lanes = <1 2>;
-					};
-				};
-			};
+			data-lanes = <1 2>;
 		};
 	};
 
 	fragment@3 {
-		target = <&i2c_csi_dsi>;
+		target = <&tc358743>;
 		__dormant__ {
-			tc358743@0f {
-				port {
-					endpoint {
-						data-lanes = <1 2 3 4>;
-					};
-				};
-			};
+			data-lanes = <1 2 3 4>;
 		};
 	};
 
